sphinx/texinputs/sphinx.sty

```diff
@@ -1879,8 +1887,18 @@
     \let\sphinxemail    \@firstofone
     \let\sphinxcrossref \@firstofone
     \let\sphinxtermref  \@firstofone
+    \def\sphinxhyphen#1{-}% the #1 is there to gobble the {}
```

Comment:
Well actually it should be without the `#1` gobbling the `{}` else hyperref seems to encode `--` into a Unicode en-dash in the PDF bookmarks, whereas a section title with `-{}-{}` does give two hyphens also in bookmarks of PDF. Will update after some additional testing.